### PR TITLE
Use URI instead of URL to enable asset loading

### LINKED ISSUE #14743: GeoJSONSource constructor for asset/file URI
To load local resources from either assets or the file system. Developers can rely on prepending their resource URI with either `asset://` or `file://`. For our GeoJsonSource component, it should be possible to use these URIs as part of the URL constructor but using something as: 

```
    try {
      markerSource = new GeoJsonSource(MARKER_SOURCE, new URL("asset://points.geojson"));
    } catch (MalformedURLException e) {
      e.printStackTrace();
    }

```

Will fail with:

```
java.net.MalformedURLException: unknown protocol
```

Users can work around this for now by using:

```java
markerSource = new GeoJsonSource(MARKER_SOURCE);
markerSource.setUrl("asset://points.geojson");
```

### PR DESCRIPTION
**Pull request by tobrun**
Closes #14743

### CODE REVIEW

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/GeoJsonSource.java [hunk 1]
```diff
@@ -247,22 +277,46 @@ public void setGeoJson(String json) {
...
     checkThread();
     setUrl(url.toExternalForm());
   }
 
+  /**
+   * Updates the uri
+   *
+   * @param url the GeoJSON FeatureCollection uri
+   */
+  public void setUri(@NonNull URI url) {
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/GeoJsonSource.java]: this javadoc could benefit of some examples: 
 - `http://`
 -  `asset://`
 - `file://`

**Comment by LukasPaczos** [comment]
Good to :white_check_mark: as soon as the javadoc examples are added.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/GeoJsonSource.java [hunk 2]
```diff
@@ -247,31 +311,101 @@ public void setGeoJson(String json) {
...
+   * <li>asset://</li>
+   * <ul>
+   * <li>load resources from the binary packaged assets folder</li>
+   * </ul>
+   *
+   * @param uri the GeoJSON FeatureCollection uri
+   */
+  public void setUri(@NonNull URI uri) {
+    checkThread();
+    setUrl(uri.toString());
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/GeoJsonSource.java]: NIT: `nativeSetUrl` or `setUri(String)` wihtout the thread check

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/VectorSource.java [hunk 3]
```diff
@@ -37,20 +38,49 @@
...
+   * @deprecated use {@link #VectorSource(String, Uri)} instead
    */
+  @Deprecated
   public VectorSource(String id, URL url) {
     this(id, url.toExternalForm());
   }
 
   /**
-   * Create a vector source from a remote url
+   * Create a vector source from a uri pointing to a TileJSON resource
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/VectorSource.java]: Missing URI protocols doc

**Pull request merged**